pyproject.toml

```diff
@@ -19,9 +19,7 @@ build-backend = "mesonpy"
 
 [project]
 name = 'pandas'
-dynamic = [
-  'version'
-]
+version = "3.0.0.dev0"
```

Comment:
Does this get rid of the commit info in the version?

If so, I'd be -1 on this.
(It may be useful for downstream to pin to a previous pandas nightly)

Reply:
No the scripts that generate the version.py file will append the git revision to the end of this

Reply:
I don't think this worked.

The filename shows up as ``3.0.0.dev0`` (at least on the sdist)

<img width="716" alt="image" src="https://github.com/pandas-dev/pandas/assets/47963215/6cfcaec3-ae38-4238-b04a-f96b771634bd">

Reply:
Maybe I misunderstood the question. If you run interactively you will see the version:

```
>>> import pandas as pd
+ /usr/local/bin/ninja
[1/1] Generating pandas/generate-version with a custom command
Saving version to pandas/_version.py
>>> pd.__version__
'3.0.0.dev0+git20240321.84fbbb0'
```

Is the sdist folder name important?

Reply:
I think so yes - pinning versions is based on the version in the sdist/wheel filename IIRC.

If you try reverting this change, do we get the git commit in the filename again?

Reply:
`_build_version/gitversion.py:L16` expects this to be `version = ...` - if you revert it back to the 'dynamic' key that no longer works

Reply:
I checked an sdist build with NumPy and they have the exact same behavior as here - is there a reason why we want to differ from them?

Reply:
Doesn't this make it so that the nightlies all basically have the same version?

(IIRC, pip figures out the version of the package from the filename, and it's hardcoded to 3.0.0dev0 for both the wheels and the sdist)

Reply:
Are we storing a longer term snapshot of the nightlies somewhere? On Anaconda I only see 5 days worth:

https://anaconda.org/scientific-python-nightly-wheels/pandas/files

NumPy only stores one day worth and no sdist:

https://pypi.anaconda.org/scientific-python-nightly-wheels/simple/numpy/